### ISSUE TITLE
Destination VM copy-exact CPUID from source VM after live-migration

### ISSUE THREAD

**Issue by likebreath** [cloud-hypervisor/cloud-hypervisor#2886]
In this way, the destination VM after live-migration will carry exactly the same CPUID information from the source VM, including advertised CPU features and CPU identification string. It can be a potential problem as the destination VM may have more supported features than being advertised to the guest workload. The correct "CPUID" information will be only available with a reboot of the destination VM.

I am not sure whether this behavior is intended. If not, we now should be safe to rely on the destination VM to create its own CPUID information given the CPUID check is added to the live-migration.

**Comment by sboeuf** [comment]
I think the behavior is intended. When we migrate the source to the destination, we want to make sure the source CPUID is a subset of the destination (this is what your PR about CPUID check is doing). Once we verified the destination VM supports a superset (or the exact same set) of CPUIDs, we don't want to "upgrade" our guest by advertising more features than the set we previously had on the source.
Imagine as a user you're running your application in a VM, and you check the CPUID available in this VM. It would be quite weird to get different output at different times because your VM has been migrated in between.
In conclusion I think the behavior during migration is correct.

Now it seems to me there's a problem with reboot since the CPUID is going to be generated from scratch while it should have been saved from the source VM. That's something we should fix, because if you take the same example as before, as a user, you're not expecting to run on an "upgraded" VM after a reboot (that might have happened after a migration).
@rbradford wdyt?

**Comment by rbradford** [comment]
After a live migration the guest at the destination must have the same CPUID information as the guest at the source. The CPUID changing after a reboot seems perfectly reasonable to me (and could even be desired.)

**Comment by sboeuf** [comment]
Well if you feel the reboot behavior is reasonable, I guess we can go ahead and close this issue :)
@likebreath, feel free to close the issue if you feel everything has been covered!

**Comment by likebreath** [comment]
Great to confirm that behavior is actually preferred!
